### ISSUE TITLE
Add output formatter compatible with mocha

### ISSUE THREAD

**Issue by piranna** [BadgeLabs/mocha-eslint#38]
Add a formater that integrates with the `mocha` output. It's said, instead of use `eslint` formaters that print directly to the console, create one that don't print anything to the console if there are some lint warnings or errors, and instead give its data to `mocha` to print them instead as mocha errors.

**Comment by IanVS** [comment]
Sounds like an awesome idea.  Are you willing to submit a PR?

**Comment by piranna** [comment]
I'm reviewing the code and seems it would be somewhat easy since it already show the output for errors. I don't think formatters makes any sense at all so I would remove them and use directly mocha output, what do you think?

**Comment by IanVS** [comment]
Hm, I'm not sure I'd go that far, as some people may want the ESLint formatted reports.  What do you think, @Turbo87?

**Comment by piranna** [comment]
Yeah, maybe not so far, I've been thinking about it and they are two different concepts, and in fact it's not possible to integrate eslint errors are mocha errors since eslint return all of them at once on the report, so maybe it's only a matter of some minor aesthetic tune-ups like removing the mocha stack (since it's meaningless here) and maybe the eslint final count, that's dependent of the formatter being used.

**Comment by Turbo87** [comment]
@piranna could you give an example of how that output should look like compared to the current state? I'm not sure I completely understand what you're suggesting yet.

**Comment by piranna** [comment]
Current output with my pull-request:

```
  eslint
    1) should have no errors in .


  1 failing

  1) eslint should have no errors in .:
     Error: Code did not pass lint rules
lib/_rules.js
  108:14  warning  Unexpected console statement  no-console

✖ 1 problem (0 errors, 1 warning)

      at Context.<anonymous> (node_modules/mocha-eslint/index.js:31:15)
```

My proposal:

```
  eslint
    1) should have no errors in .


  1 failing

  1) eslint should have no errors in .:
     Error: Code did not pass lint rules
     lib/_rules.js
       108:14  warning  Unexpected console statement  no-console
```

**Comment by piranna** [comment]
In fact, I would also remove the line `Error: Code did not pass lint rules` too, since we are already showing the error.

**Comment by Turbo87** [comment]
to be honest I don't really see a strong advantage here and I'm not quite sure how you're planning to implement this

**Comment by piranna** [comment]
I'm thinking about indenting the output so it get aligned with the mocha report, and I think the stack can be removed by reporting the output using the `done()` callback as it's done with async tests instead of throwing an error (that's what's causing the stacktrace, by the way).

**Comment by Turbo87** [comment]
I think the indenting should actually be done inside the console reporter of mocha. It already indents the first line, it "just" needs to be modified to indent all lines. I guess we could also set the stacktrace of the error to `''` and then throw it. I'm not sure why using `done()` would remove the stacktrace?

**Comment by piranna** [comment]
> I think the indenting should actually be done inside the console reporter of mocha. It already indents the first line, it "just" needs to be modified to indent all lines.

I was thinking about using [indent-string](https://www.npmjs.com/package/indent-string) so we can be able to indent all the lines "as a block", making both eslint and mocha agnostic about that since we are directly printing the output of eslint.


> I guess we could also set the stacktrace of the error to '' and then throw it. I'm not sure why using done() would remove the stacktrace?

If simply by setting the stack to '' works, that's ok for me :-)

**Comment by piranna** [comment]
> If simply by setting the stack to '' works, that's ok for me :-)

Ok, it worked :-) Only that now it left three empty lines on the bottom, but at least there's no confusing data on the output.

**Comment by piranna** [comment]
Et voila! :-D

```
[piranna@Natalie:~/Séntisis/sentisis-linguist]
 (develop) > npm test

> sentisis-linguist@0.0.0 test /Users/piranna/Séntisis/sentisis-linguist
> mocha



  linguist
    ✓ get response (69ms)

  eslint
    1) should have no errors in .


  1 passing (3s)
  1 failing

  1) eslint should have no errors in .:
     Code did not pass lint rules
     lib/_rules.js
       108:14  warning  Unexpected console statement  no-console

     ✖ 1 problem (0 errors, 1 warning)
     
  



npm ERR! Test failed.  See above for more details.
```

**Comment by Turbo87** [comment]
>> I think the indenting should actually be done inside the console reporter of mocha. It already indents the first line, it "just" needs to be modified to indent all lines.

> I was thinking about using indent-string so we can be able to indent all the lines "as a block", making both eslint and mocha agnostic about that since we are directly printing the output of eslint.

again: properly indenting the error message is a responsibility of mocha itself and its console reporter, but shouldn't be done by mocha-eslint

**Comment by piranna** [comment]
> again: properly indenting the error message is a responsibility of mocha itself and its console reporter, but shouldn't be done by mocha-eslint

I've review mocha source code and [it's only indenting the stack](https://github.com/mochajs/mocha/blob/9ae6a85b6ba50ea415affc88456d469d6915a157/lib/reporters/base.js#L219), for messages it only indent the [first](https://github.com/mochajs/mocha/blob/9ae6a85b6ba50ea415affc88456d469d6915a157/lib/reporters/base.js#L166) [line](https://github.com/mochajs/mocha/blob/9ae6a85b6ba50ea415affc88456d469d6915a157/lib/reporters/base.js#L209), that's the one that has the `Code did not pass lint rules` text and ignoring if there are break lines in the main text, that's what we are interested on indenting here to align with the mocha report output.

**Comment by Turbo87** [comment]
@piranna I'm not sure you understand what I'm suggesting. Instead of changing `mocha-eslint` you should instead change `mocha` to always indent error messages correctly.

if you solve it just for this plugin others will still have the same problem. if you solve it in mocha itself then it will be solved for everyone.

**Comment by piranna** [comment]
Good point, didn't though about it... [I've open an issue on mocha repo](https://github.com/mochajs/mocha/issues/2159) to request for comments about this.

**Comment by IanVS** [comment]
@piranna, What do your changes look like if you don't do the indenting?  If it adds value, we can implement it while https://github.com/mochajs/mocha/issues/2159 is being discussed/addressed.

**Comment by piranna** [comment]
> @piranna, What do your changes look like if you don't do the indenting? If it adds value, we can implement it while mochajs/mocha#2159 is being discussed/addressed.

@Turbo87 showed a good example on the `mocha` repo.
